tracer/src/Datadog.Trace/Activity/Handlers/AzureServiceBusActivityHandler.cs

```diff
@@ -24,11 +28,69 @@ public void ActivityStarted<T>(string sourceName, T activity)
         {
             var tags = Tracer.Instance.CurrentTraceSettings.Schema.Client.CreateAzureServiceBusTags();
             ActivityHandlerCommon.ActivityStarted(sourceName, activity, tags: tags, out var activityMapping);
-            // TODO: Update the Activity's traceparent and/or tracestate strings here
         }
 
         public void ActivityStopped<T>(string sourceName, T activity)
             where T : IActivity
-            => ActivityHandlerCommon.ActivityStopped(sourceName, activity);
+        {
+            var dataStreamsManager = Tracer.Instance.TracerManager.DataStreamsManager;
+            if (Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationId.AzureServiceBus)
+                && dataStreamsManager.IsEnabled
+                && activity.Instance is not null
+                && activity.OperationName == "Message"
+                && AzureServiceBusCommon.ActiveMessageProperties is var applicationProperties
+                && applicationProperties is not null)
+            {
+                // Adding DSM to the send operation of IReadOnlyCollection<ServiceBusMessage>|ServiceBusMessageBatch - Step Three:
+                // If we can retrieve the active message properties object stored in the AsyncLocal field,
+                // then we can retrieve the active message object using our mapping. With access to the message
+                // object, we can accurately calculate the payload size for the DataStreamsCheckpoint
+
+                string key;
+                if (activity is IW3CActivity w3cActivity)
+                {
+                    key = w3cActivity.TraceId + w3cActivity.SpanId;
+                }
+                else
+                {
+                    key = activity.Id;
+                }
+
+                if (ActivityHandlerCommon.ActivityMappingById.TryRemove(key, out ActivityMapping activityMapping)
+                    && activityMapping.Scope?.Span is Span span)
+                {
+                    // Copy over the data to our Span object so we can do an efficient tags lookup
+                    OtlpHelpers.UpdateSpanFromActivity(activity, span);
+
+                    string namespaceString = span.Tags.GetTag("messaging.destination.name");
+                    long? payloadSize = null;
+                    if (AzureServiceBusCommon.TryGetMessage(applicationProperties, out var message)
+                        && message.TryDuckCast<IServiceBusMessage>(out var serviceBusMessage))
+                    {
+                        payloadSize = AzureServiceBusCommon.GetMessageSize(serviceBusMessage);
+                    }
+
+                    var edgeTags = string.IsNullOrEmpty(namespaceString)
+                        ? new[] { "direction:out", "type:azureservicebus" }
+                        : new[] { "direction:out", $"topic:{namespaceString}", "type:azureservicebus" };
```

Comment:
For Azure Service Bus, you can either:
1. Send to a Queue and receive from that Queue
2. Send to a Topic (which acts like an exchange) and receive from a Subscription

Ideally, I would do the following:
- Queue: `topic:<queueName>`
- Topic: `exchange:<topicName>`

If I can't figure out a way to disambiguate the two, what should I do? When you send a message using the library, it treats your string input as just a `string EntityPath`, so I don't know if we can actually tell on the client side if the message is going to a queue or a topic. I'll keep digging to see if it's possible to determine but at the moment I'm not sure if we will be able to

Reply:
See this docs link for more info: https://learn.microsoft.com/en-us/dotnet/api/azure.messaging.servicebus.servicebussender?view=azure-dotnet